src/transformers/adapters/peft_mixin.py

```diff
@@ -0,0 +1,258 @@
+                "adapter model."
+            )
+
+        loaded_peft_config = PeftConfig.from_pretrained(
+            peft_model_id,
+            revision=revision,
+            use_auth_token=use_auth_token,
+            commit_hash=commit_hash,
+        )
+
+        # Replace the adapter with the loaded adapter
+        inject_adapter_in_model(loaded_peft_config, self, adapter_name)
+
+        adapter_state_dict = load_peft_weights(
+            peft_model_id,
+            revision=revision,
+            use_auth_token=use_auth_token,
+        )
+
+        # We need to pre-process the state dict to remove unneeded prefixes - for backward compatibility
+        processed_adapter_state_dict = {}
+        for key, value in adapter_state_dict.items():
+            if "base_model.model" in key:
+                new_key = key.replace("base_model.model.", "")
+            else:
+                new_key = key
+            processed_adapter_state_dict[new_key] = value
+
+        # Load state dict
+        incompatible_keys = set_peft_model_state_dict(self, processed_adapter_state_dict, adapter_name)
+
+        if incompatible_keys is not None:
+            # check only for unexpected keys
+            if hasattr(incompatible_keys, "unexpected_keys") and len(incompatible_keys.unexpected_keys) > 0:
+                logger.warning(
+                    f"Loading adapter weights from {peft_model_id} led to unexpected keys not found in the model: "
+                    f" {incompatible_keys.unexpected_keys}. "
+                )
+
+        # @pacman100 why this was needed?
+        if (
+            (getattr(self, "hf_device_map", None) is not None)
+            and (len(set(self.hf_device_map.values()).intersection({"cpu", "disk"})) > 0)
+            and len(self.peft_config) == 1
+        ):
+            self._dispatch_accelerate_model(
+                device_map=device_map, max_memory=max_memory, offload_dir=offload_dir, offload_index=offload_index
+            )
+
+    def add_adapter(
+        self,
+        adapter_config,
+        adapter_name: Optional[str] = "default",
+    ) -> None:
+        r"""
+        Adds a fresh new adapter to the current model for training purpose.
+        """
+        requires_backends(self.add_adapter, "peft")
+
+        from peft import PeftConfig, inject_adapter_in_model
+
+        if not self._hf_peft_config_loaded:
+            self._hf_peft_config_loaded = True
+        elif adapter_name in self.peft_config:
+            raise ValueError(f"Adapter with name {adapter_name} already exists. Please use a different name.")
+
+        if not isinstance(adapter_config, PeftConfig):
+            raise ValueError(
+                f"adapter_config should be an instance of PeftConfig. Got {type(adapter_config)} instead."
+            )
+
+        inject_adapter_in_model(adapter_config, self, adapter_name)
+
+    def set_adapter(self, adapter_name: str) -> None:
```

Comment:
What is an example of an `"adapter_name"`? 
Does the user decide themselves what the adapter name should be?

Reply:
should there also be a "remove_adapter" function?

Reply:
yes users can decide what the adapter name should be when loading an adapter or adding an adapter, and switch between them using `set_adapter`
I am not sure we should add a `remove_adapter` method as we now have a `disable_adapters` method here: [2345681](https://github.com/huggingface/transformers/pull/25077/commits/234568191e3fbd8364b957c0cc7cc1e5a4424370). 
 What kind of usage would you see for `remove_adapter`?

Reply:
Ok!

`remove_adapter` could be useful to make the model lighter again and eventually completely remove all adapters. But probably better in a next iteration.

Reply:
OK having it in the next iteration sounds good!